File: 1614-Maximum-Nesting-Depth-of-the-Parentheses/1614-Maximum-Nesting-Depth-of-the-Parentheses.go
func maxDepth(s string) int {
    left := 0
	right := 0
	max := 0

	for _, c := range s {
		if c == '(' {
			left++
		} else if c == ')' {
			right++
		}
		sum := left - right
		if sum > max {
			max = sum
		}
	}

	return max
    
}

File: 1614-Maximum-Nesting-Depth-of-the-Parentheses/1614-Maximum-Nesting-Depth-of-the-Parentheses.java
class Solution {
    public int maxDepth(String s) {
        int left = 0;
        int right = 0;
        int max = 0;
        for (char c : s.toCharArray()) {
            if (c == '(') {
                left++;
            } else if (c == ')') {
                right++;
            }
            int sum = left - right;
            max = Math.max(max, sum);
        }
        return max;
        
    }
}
